### PR TITLE
Improved latitude and longitude validation, and fixed help function

### PR DESCRIPTION
**Pull request by DMWhiteley**
For issue #75 

Two changes:
1. Fixed a ClassCastException when calling main -help
2. Improved latitude and longitude validators based on data.world feedback report:
   a. More robust handling of non-decimal values
   b. Added validation to eliminate invalid decimal values (i.e. longitudes over/under 180/-180)
   c. Added validation tests for latitudes and longitudes.

### CODE REVIEW

File: src/main/java/org/wikivoyage/listings/validators/LongitudeValidator.java [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+      	//Sanitise the longitude to see if it contained invalid chars. Not trying to fix, just flag as invalid.
+      	// This is more robust than just relying on the NFE because we explicitly know what a longitude SHOULD contain.
+      	String sanitisedLongitude = longitude.replaceAll("[^0-9.-]","");
+      	if (!sanitisedLongitude.equals(longitude)) {
+          	poi.add(ValidationIssue.INVALID_LONGITUDE);
+          }
+          else {
+          	//Otherwise try to parse this as a valid longitude.
+              try {
+                  float latitudeF = Float.parseFloat(longitude);
```
Comment (matevarga) [inline src/main/java/org/wikivoyage/listings/validators/LongitudeValidator.java]: You might want to change the variable name to match the actual meaning. 👍 Plus this validator shares most of the code with the other -- any chance you can refactor?

File: src/main/java/org/wikivoyage/listings/validators/LatitudeValidator.java [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
-        if (poi.getLatitude() != null && !poi.getLatitude().equals("")) {
+    	String latitude = poi.getLatitude();
+    	if (StringUtils.isEmpty(latitude)) {
+    		//Do nothing; not invalid but no point parsing
+    		return;
+    	}
+    	
+    	//Sanitise the latitude to see if it contained invalid chars. Not trying to fix, just flag as invalid.
+    	// This is more robust than just relying on the NFE because we explicitly know what a latitude SHOULD contain.
+    	String sanitisedLatitude = latitude.replaceAll("[^0-9.-]","");
```
Comment (matevarga) [inline src/main/java/org/wikivoyage/listings/validators/LatitudeValidator.java]: Won't the pattern be recompiled on each usage?

File: src/main/java/org/wikivoyage/listings/validators/LongitudeValidator.java (hunk 1 above)
Comment (DMWhiteley) [reply src/main/java/org/wikivoyage/listings/validators/LongitudeValidator.java]: *groan* Good spot on the variable name. Caused by copy-pasting, so a good sign that refactoring is needed, should be easy enough.

File: src/main/java/org/wikivoyage/listings/validators/LatitudeValidator.java (hunk 2 above)
Comment (DMWhiteley) [reply src/main/java/org/wikivoyage/listings/validators/LatitudeValidator.java]: Declared statically now so it'll only be compiled on instantiation

**Comment by DMWhiteley** [comment]
@nicolas-raoul , any chance to check this PR yet? Thanks!

**Comment by nicolas-raoul** [comment]
Thanks for the PR!
I will check asap 🙂

**Pull request closed without merging**